demos/python_demos/object_detection_demo_faceboxes/detector.py

```diff
@@ -0,0 +1,202 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import itertools
+import math
+import numpy as np
+import os
+import cv2
+from collections import namedtuple
+
+class Detector(object):
+    def __init__(self, ie, model_path, device='CPU', threshold=0.5):
+        model = ie.read_network(model=model_path, weights=os.path.splitext(model_path)[0] + '.bin')
+
+        assert len(model.inputs) == 1, "Expected 1 input blob"
+        assert len(model.outputs) == 2, "Expected 2 output blobs"
+
+        self._input_layer_name = next(iter(model.inputs))
+        self._output_layer_names = sorted(model.outputs)
+
+        assert model.outputs[self._output_layer_names[0]].shape[0] == \
+               model.outputs[self._output_layer_names[1]].shape[0], "Expected the same dimension for boxes and scores"
+        assert model.outputs[self._output_layer_names[0]].shape[1] == 4, "Expected 4-coordinate boxes"
+        assert model.outputs[self._output_layer_names[1]].shape[1] == 2, "Expected 2-class scores(background, face)"
```

Comment:
It looks indices are biased, 0 dimension stays for batch size, not detected objects.

Reply:
Fixed